aspnetcore/fundamentals/middleware/extensibility.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Factory-based middleware activation in ASP.NET Core
+author: guardrex
+description: Learn how to use strongly-typed middleware with a factory-based activation implementation in ASP.NET Core.
+ms.author: riande
+manager: wpickett
+ms.custom: mvc
+ms.date: 01/23/2018
+ms.topic: article
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: fundamentals/middleware/extensibility
+---
+# Factory-based middleware activation in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex)
+
+[IMiddlewareFactory](/dotnet/api/microsoft.aspnetcore.http.imiddlewarefactory)/[IMiddleware](/dotnet/api/microsoft.aspnetcore.http.imiddleware) is an extensibility point for [middleware](xref:fundamentals/middleware/index) activation.
+
+`UseMiddleware` extension methods check if a middleware's registered type implements `IMiddleware`. If it does, the `IMiddlewareFactory` instance registered in the container is used to resolve the `IMiddleware` implementation instead of using the convention-based middleware activation logic. The factory is registered as a scoped service with the app's service container.
+
+Benefits:
+
+* Activation per request (injection of scoped services)
+* Strong typing of middleware
+
+`IMiddleware` is activated per request, so scoped services can be injected into the constructor.
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/middleware/extensibility/sample) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+The sample app demonstrates:
+
+* Conventionally-activated middleware.
+* Middleware activated by an `IMiddlewareFactory` registered in an [Ninject](http://www.ninject.org/) container.
```

Comment:
~~registered in an~~
registered in a

Reply:
First syllable is "in" (as in "inject" with a silent "n"), so "an" here.

Reply:
Good to know. I've always heard it pronounced with the leading "N".

Reply:
I meant the first "N" in "Ninject" would be silent, but YOU are correct ... "nin-ject," so the first syllable *really does* start with a consonant.